src/components/view/index.tsx

```diff
@@ -46,106 +40,90 @@ export interface ViewProps extends Omit<RNViewProps, 'style'>, ContainerModifier
-
-  constructor(props: PropsTypes) {
-    super(props);
-
-    this.Container = props.useSafeArea && Constants.isIOS ? SafeAreaView : RNView;
-    if (props.reanimated) {
-      this.Container = Reanimated.createAnimatedComponent(this.Container);
-    } else if (props.animated) {
-      this.Container = Animated.createAnimatedComponent(this.Container);
-    }
-
-    this.state = {
-      ready: !props.renderDelay
-    };
-  }
+function View(props: ViewProps, ref: any) {
+  const themeProps = useThemeProps(props, 'View');
+  const {
+    renderDelay,
+    style,
+    // (!) extract left, top, bottom... props to avoid passing them on Android
+    /* eslint-disable */
+    left,
+    top,
+    right,
+    bottom,
+    flex: propsFlex,
+    /* eslint-enable */
+    inaccessible,
+    useSafeArea,
+    animated,
+    reanimated,
+    children,
+    ...others
+  } = themeProps;
+  const {backgroundColor, borderRadius, paddings, margins, alignments, flexStyle, positionStyle} = useModifiers(themeProps,
+    modifiersOptions);
+  const [ready, setReady] = useState(!renderDelay);
 
-  componentDidMount() {
-    const {renderDelay} = this.props;
+  useEffect(() => {
     if (renderDelay) {
       setTimeout(() => {
-        this.setState({ready: true});
+        setReady(true);
       }, renderDelay);
     }
-  }
+  }, []);
 
-  // TODO: do we need this?
-  setNativeProps(nativeProps: any) {
-    //@ts-ignore
-    this._root.setNativeProps(nativeProps); // eslint-disable-line
-  }
+  const ViewContainer = useMemo(() => {
+    const container = useSafeArea && Constants.isIOS ? SafeAreaView : RNView;
 
-  render() {
-    if (!this.state.ready) {
-      return null;
+    if (reanimated) {
+      return Reanimated.createAnimatedComponent(container);
+    } else if (animated) {
+      return Animated.createAnimatedComponent(container);
     }
 
-    // (!) extract left, top, bottom... props to avoid passing them on Android
-    // eslint-disable-next-line
-    const {
-      modifiers,
-      style,
-      /* eslint-disable */
-      left,
-      top,
-      right,
-      bottom,
-      flex: propsFlex,
-      /* eslint-enable */
-      forwardedRef,
-      inaccessible,
-      ...others
-    } = this.props;
-    const {backgroundColor, borderRadius, paddings, margins, alignments, flexStyle, positionStyle} = modifiers;
-    const Element = this.Container;
-    return (
-      <Element
-        accessibilityElementsHidden={inaccessible}
-        importantForAccessibility={inaccessible ? 'no-hide-descendants' : undefined}
-        {...others}
-        style={[
-          backgroundColor && {backgroundColor},
-          borderRadius && {borderRadius},
-          flexStyle,
-          positionStyle,
-          paddings,
-          margins,
-          alignments,
-          style
-        ]}
-        ref={forwardedRef}
-      >
-        {this.props.children}
-      </Element>
-    );
+    return container;
+  }, [useSafeArea, animated, reanimated]);
```

Comment:
I can see `useSafeArea` changing, but probably not the other two

Reply:
I agree, And if they won't change there's no harm having them here (:

Reply:
It might have some performance implications, but they are probably minor